src/main/scala/mesosphere/util/monitor/HeartbeatMonitor.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.util.monitor
+
+import javax.inject.Inject
+
+import akka.actor._
+import akka.event.LoggingReceive
+import scala.concurrent.duration._
+
+// HeartbeatActor monitors the heartbeat of some process and executes handlers for various conditions.
```

Comment:
There's kind of two pieces here: one is to put it in the marathon package structure, the second is that it seems to be pretty specific to mesos scheduler's requirements for heartbeating, no? If not, maybe it should be used in the health check infrastructure, for example? If the former, I support the tight coupling.